Library/Homebrew/shims/super/cc

```diff
@@ -292,6 +292,8 @@ class Cmd
     args.concat(optflags) unless runtime_cpu_detection?
     args.concat(archflags)
     args << "-std=#{@arg0}" if /c[89]9/.match?(@arg0)
+    # Add -nostdinc when building against glibc@2.13 to avoid mixing system and brewed glibc headers.
+    args << "-nostdinc" if @deps.include?("glibc@2.13")
```

Comment:
Looks like this is gated to not run if we're not doing arg refurbishment (line 287).

I argue we actually want to do this unconditionally. I think it probably should also be in `cppflags` like the other `-isystem` flags.

Reply:
That makes sense.  I tried this and it fixed the issue where the system headers were still being included seen here: https://github.com/Homebrew/homebrew-portable-ruby/pull/156#issuecomment-1190303615.